src/ui/server/controllers/insurance/your-buyer/outstanding-or-overdue-payments/save-and-back/index.test.ts

```diff
@@ -0,0 +1,125 @@
+import { FIELD_IDS } from '..';
+import { post } from '.';
+import { ROUTES } from '../../../../../constants';
+import INSURANCE_FIELD_IDS from '../../../../../constants/field-ids/insurance';
+import constructPayload from '../../../../../helpers/construct-payload';
+import generateValidationErrors from '../validation';
+import mapAndSave from '../../map-and-save/buyer-trading-history';
+import { Request, Response } from '../../../../../../types';
+import { mockReq, mockRes, mockSpyPromiseRejection } from '../../../../../test-mocks';
+
+const { OUTSTANDING_PAYMENTS, FAILED_PAYMENTS, TOTAL_AMOUNT_OVERDUE } = INSURANCE_FIELD_IDS.YOUR_BUYER;
+
+const { INSURANCE_ROOT, ALL_SECTIONS, PROBLEM_WITH_SERVICE } = ROUTES.INSURANCE;
+
+describe('controllers/insurance/your-buyer/outstanding-or-overdue-payments/save-and-back', () => {
+  let req: Request;
+  let res: Response;
+
+  let updateMapAndSave = jest.fn(() => Promise.resolve(true));
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+
+    mapAndSave.buyerTradingHistory = updateMapAndSave;
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  const validBody = {
+    [OUTSTANDING_PAYMENTS]: 'true',
+    [FAILED_PAYMENTS]: 'false',
+  };
+
+  describe('when there are no validation errors', () => {
+    it('should redirect to all sections page', async () => {
+      req.body = validBody;
+
+      await post(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`${INSURANCE_ROOT}/${req.params.referenceNumber}${ALL_SECTIONS}`);
+    });
+
+    it('should call mapAndSave.buyerTradingHistory once with data from constructPayload function', async () => {
+      req.body = validBody;
+
+      await post(req, res);
+
+      expect(updateMapAndSave).toHaveBeenCalledTimes(1);
+
+      const payload = constructPayload(req.body, FIELD_IDS);
+      const validationErrors = generateValidationErrors(payload);
+
+      expect(updateMapAndSave).toHaveBeenCalledWith(payload, res.locals.application, validationErrors);
+    });
+  });
+
+  describe('when there are validation errors', () => {
+    it('should redirect to all sections page', async () => {
+      req.body = validBody;
+
+      await post(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`${INSURANCE_ROOT}/${req.params.referenceNumber}${ALL_SECTIONS}`);
+    });
+
+    it('should call mapAndSave.yourBuyer once with data from constructPayload function', async () => {
```

Comment:
Seems we have duplicate `it('should call mapAndSave ...` tests here

Reply:
one has validation errors and one does not? the rest in this directory are the same